Common/Brokerages/FxcmBrokerageModel.cs

```diff
@@ -190,6 +190,18 @@ public override ISlippageModel GetSlippageModel(Security security)
             return new ConstantSlippageModel(0);
         }
 
+        /// <summary>
+        /// Gets a new settlement model for the security
+        /// </summary>
+        /// <param name="security">The security to get a settlement model for</param>
+        /// <returns>The settlement model for this brokerage</returns>
+        public override ISettlementModel GetSettlementModel(Security security)
+        {
+            return security.Type == SecurityType.Cfd
+                ? new AccountCurrencyImmediateSettlementModel() :
+                (ISettlementModel)new ImmediateSettlementModel();
```

Comment:
I view this as pending work for issue #110 - ideally please make the `ImmediateSettlementModel` support generic account currencies.